spec/models/sponsor_spec.rb

```diff
@@ -328,6 +328,24 @@
       expect(request_fulfilled_sponsor.eligible_for_sponsorship?).to eq false
     end
 
+    describe '#sponsorship_changed!', :mytest => true do
+
+      before(:each) do
+        allow(active_sponsor).to receive(:update_request_fulfilled!)
+        allow(active_sponsor).to receive(:update_active_sponsorship_count!)
+        active_sponsor.sponsorship_changed!
+      end
+
+      it 'should call #update_request_fulfilled!' do
+        expect(active_sponsor).to have_received(:update_request_fulfilled!)
+      end
+      
+      it 'should call #update_active_sponsorship_count!' do
+        expect(active_sponsor).to have_received(:update_active_sponsorship_count!)
+      end
```

Comment:
Can you try without the `allow` stubs? The `expect to have_received` should cover it. If that fails, I'd prefer to change the `allow` to `expect` on L1&2 and remove the two last lines to avoid duplication.